projects/plugins/boost/app/lib/critical-css/Critical_CSS_State.php

```diff
@@ -107,37 +107,11 @@ private function get_state_transient() {
 		);
 	}
 
-	public function get_generation_status() {
-		// Todo: created time may be different in browser.
-		$state  = $this->get_state_transient();
-		$status = $this->get_providers_status( $state['sources'] );
-
-		return array(
-			'created'   => $state['created'],
-			'updated'   => $state['updated'],
-			'completed' => $status['completed'],
-			'total'     => $status['total'],
-			'pending'   => $status['pending'],
-		);
-	}
-
-	private function get_providers_status( $providers ) {
-		$completed = 0;
-		$pending   = false;
-
-		foreach ( $providers as $provider ) {
-			if ( self::SUCCESS === $provider['status'] ) {
-				$completed++;
-			} elseif ( self::REQUESTING === $provider['status'] ) {
-				$pending = true;
-			}
+	public function maybe_set_status() {
+		if ( $this->get_total_providers_count() === $this->get_processed_providers_count() ) {
```

Comment:
This actually looks like a correct way to do it. However, the current implementation shows a primary error set in case of all the providers fail with non-showstopper errors and the overall status is `success`. I think we can refactor this with the scope of a new ticket.